pkg/trace/writer/zstd.go

```diff
@@ -0,0 +1,24 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build zstd
+
+package writer
+
```

Comment:
Build tags don't work with OTel Collector. Please make zstd writer an interface than can have noop implementation.

Reply:
If no build tag is passed, `zstd` will not be available during the build process. (and we will fall back to gzip). How is this incompatible with OTel Collector?